tempesta_fw/str.c

```diff
@@ -271,6 +274,42 @@ tfw_strcpy(TfwStr *dst, const TfwStr *src)
 }
 EXPORT_SYMBOL(tfw_strcpy);
 
+/**
+ * Function for copying of TfwStr descriptors only.
+ */
+int
+tfw_strcpy_desc(TfwStr *dst, TfwStr *src)
+{
+	TfwStr *c1, *c2, *end;
+	int src_num = TFW_STR_CHUNKN(src);
+	int dst_num = TFW_STR_CHUNKN(dst);
+
+	/*
+	 * Only TfwStr descriptors with identical complexity
+	 * and chunks number are allowed.
+	 */
+	if (unlikely(src_num != dst_num))
+		return -E2BIG;
+
+	c2 = dst_num ? dst->ptr : dst;
+	TFW_STR_FOR_EACH_CHUNK(c1, src, end) {
+		c2->ptr = c1->ptr;
+		c2->skb = c1->skb;
+		c2->eolen = c1->eolen;
+		c2->len = c1->len;
+		c2->flags = c1->flags;
```

Comment:
Perhaps, a simple `c2 = c1` would suffice here?

Reply:
Yes, missed that.
Corrected.